frontend/src/utils/board/canvasUtils.ts

```diff
@@ -0,0 +1,97 @@
+import { fabric } from "fabric";
+import { CanvasRef } from "@/interfaces/canvas-context";
+import { CanvasObjectTypes } from "@/enums/CanvasObjectTypes";
+
+export const getSelectedObjectStyles = (canvas: fabric.Canvas | null): object | null => {
+  if (canvas) {
+    const activeObject = canvas.getActiveObject();
+    if (activeObject) {
+      return activeObject.toObject();
+    }
+  }
+  return null;
+};
+
+export const setSelectedObjectStyles = (canvas: fabric.Canvas | null, styles: object): void => {
+  if (!canvas) {
+    return;
+  }
+  const activeObjects = canvas.getActiveObjects();
+  activeObjects.forEach((obj) => {
+    obj.set(styles);
+  });
+  canvas.requestRenderAll();
+};
+
+export const initializeCanvas = (canvasRef: CanvasRef): fabric.Canvas | null => {
+  if (canvasRef.current) {
+    return new fabric.Canvas(canvasRef.current, {
+      selection: true,
+    });
+    // i am not sure if this is needed, i will leave this here if there will be a problem
+    //newCanvas.on("before:render", () => (newCanvas.selection = false));
+    //newCanvas.on("after:render", () => (newCanvas.selection = true));
+  }
+  return null;
+};
+
+export const toggleDrawingMode = (canvas: fabric.Canvas | null, isDrawingMode: boolean): void => {
+  if (canvas) {
+    canvas.isDrawingMode = isDrawingMode;
+  }
+};
+
+/**
+ * This function scales the passed `fabricjs` object.
+ * It reads the values of `scaleX` and `scaleY` properties and applies them to the `width` and `height` properties
+ * by simply multiplying them accordingly. As of result the `scaleX` and `scaleY` properties are set to 1.
+ *
+ * This function supports partialy erased objects (takes into account the `eraser` property).
+ * @param obj
+ */
+export const updateDimensions = (obj: any): void => {
+  if (!obj) {
+    return;
+  }
+  if (
+    obj.type === CanvasObjectTypes.Rect ||
+    obj.type === CanvasObjectTypes.Circle ||
+    obj.type === CanvasObjectTypes.Line
+  ) {
+    const initialObjScaleX = obj.scaleX;
+    const initialObjScaleY = obj.scaleY;
+    const scaledWidth = Math.round(obj.width * initialObjScaleX);
+    const scaledHeight = Math.round(obj.height * initialObjScaleY);
+    const scaledRadius = obj.radius ? Math.round(obj.radius * Math.max(obj.scaleX, obj.scaleY)) : undefined;
+    // scale the object itself
+    obj.set({
+      width: scaledWidth,
+      height: scaledHeight,
+      radius: scaledRadius,
+      scaleX: 1,
+      scaleY: 1,
+    });
+
+    // scale the erased parts
+    if (obj.eraser) {
+      obj.eraser.scaleX *= initialObjScaleX;
+      obj.eraser.scaleY *= initialObjScaleY;
+    }
+  }
+  obj.setCoords();
+};
+
+//todo: make use of this zoom function
```

Comment:
I think we can make a separate task out of it. Or even better - we could write a user story, which would precisely determine what we actually want to achieve.

Reply:
yes, i just wanted  to write it somewhere, we have lots of work now, so this feature won't be implemented in a week, but yea, we can create a task and move to backlog